array-manipulation-3/src/flatten.js

```diff
@@ -0,0 +1,27 @@
+/* exported flatten */
+// create a single bundle
+// take input bundle of values, and add each value to single bundle
+// if input bundle is empty, return single empty bundle
+// output single bundle of values
+
+// flatten([['foo', 'bar'], ['baz', 'qux']]);
+// // -> ["foo", "bar", "baz", "qux"]
+
+function flatten(array) {
+  var singleBundle = []; // ['foo', 'bar']
+
+  if (array.length === 0) {
+    return singleBundle;
+  }
```

Comment:
This shouldn't be necessary. If the array is empty, the loop will never start, so you are left with the same result.